### PR TITLE
refactor(elb/data_source): fix elb data source lint issues

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  fix elb data source lint issues

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  fix elb data source lint issues
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/elb/' TESTARGS='-run TestAccDataSourceELbCertificateV3_basic' 
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/elb/ -v -run TestAccDataSourceELbCertificateV3_basic -timeout 360m -parallel 4 
=== RUN   TestAccDataSourceELbCertificateV3_basic 
--- PASS: TestAccDataSourceELbCertificateV3_basic (10.50s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/elb       10.552s
```

### CODE REVIEW

File: huaweicloud/services/elb/data_source_huaweicloud_elb_certificate.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func dataSourceELbCertificateV3Read(_ context.Context, d *schema.ResourceData, m
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/data_source_huaweicloud_elb_certificate.go]: config is same with the package name, it is recommended to rename it.

File: huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go [hunk 2]
```diff
@@ -106,7 +106,7 @@ func resourceCertificateV3Create(ctx context.Context, d *schema.ResourceData, me
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go [hunk 3]
```diff
@@ -135,25 +135,25 @@ func resourceCertificateV3Read(_ context.Context, d *schema.ResourceData, meta i
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go [hunk 4]
```diff
@@ -167,7 +167,7 @@ func resourceCertificateV3Update(ctx context.Context, d *schema.ResourceData, me
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go [hunk 5]
```diff
@@ -202,7 +202,7 @@ func resourceCertificateV3Delete(_ context.Context, d *schema.ResourceData, meta
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_ipgroup.go [hunk 6]
```diff
@@ -76,26 +76,26 @@ func ResourceIpGroupV3() *schema.Resource {
...
-		IpList = append(IpList, ipListOpts)
+		ipLists = append(ipLists, ipListOpt)
 	}
 
-	return IpList
+	return ipLists
 }
 
 func resourceIpGroupV3Create(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_ipgroup.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_ipgroup.go [hunk 7]
```diff
@@ -121,24 +121,24 @@ func resourceIpGroupV3Read(_ context.Context, d *schema.ResourceData, meta inter
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_ipgroup.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_ipgroup.go [hunk 8]
```diff
@@ -159,7 +159,7 @@ func resourceIpGroupV3Update(ctx context.Context, d *schema.ResourceData, meta i
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_ipgroup.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_ipgroup.go [hunk 9]
```diff
@@ -188,12 +188,12 @@ func resourceIpGroupV3Delete(_ context.Context, d *schema.ResourceData, meta int
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_ipgroup.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_l7policy.go [hunk 10]
```diff
@@ -67,9 +67,9 @@ func ResourceL7PolicyV3() *schema.Resource {
 
 func resourceL7PolicyV3Create(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_l7policy.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_l7policy.go [hunk 11]
```diff
@@ -100,12 +100,12 @@ func resourceL7PolicyV3Create(ctx context.Context, d *schema.ResourceData, meta
 
 func resourceL7PolicyV3Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_l7policy.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_l7policy.go [hunk 12]
```diff
@@ -128,9 +128,9 @@ func resourceL7PolicyV3Read(_ context.Context, d *schema.ResourceData, meta inte
 
 func resourceL7PolicyV3Update(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_l7policy.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_l7policy.go [hunk 13]
```diff
@@ -165,19 +165,19 @@ func resourceL7PolicyV3Update(ctx context.Context, d *schema.ResourceData, meta
 
 func resourceL7PolicyV3Delete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_l7policy.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_l7rule.go [hunk 14]
```diff
@@ -81,12 +81,12 @@ func ResourceL7RuleV3() *schema.Resource {
 
 func resourceL7RuleV3Create(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_l7rule.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_l7rule.go [hunk 15]
```diff
@@ -116,22 +116,22 @@ func resourceL7RuleV3Create(ctx context.Context, d *schema.ResourceData, meta in
 
 func resourceL7RuleV3Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_l7rule.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_l7rule.go [hunk 16]
```diff
@@ -145,12 +145,12 @@ func resourceL7RuleV3Read(_ context.Context, d *schema.ResourceData, meta interf
 
 func resourceL7RuleV3Update(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_l7rule.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_l7rule.go [hunk 17]
```diff
@@ -178,20 +178,20 @@ func resourceL7RuleV3Update(ctx context.Context, d *schema.ResourceData, meta in
 
 func resourceL7RuleV3Delete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_l7rule.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_listener.go [hunk 18]
```diff
@@ -164,10 +164,10 @@ func resourceListenerV3Create(ctx context.Context, d *schema.ResourceData, meta
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_listener.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_listener.go [hunk 19]
```diff
@@ -330,130 +330,138 @@ func resourceListenerV3Update(ctx context.Context, d *schema.ResourceData, meta
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_listener.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_listener.go [hunk 20]
```diff
@@ -330,130 +330,138 @@ func resourceListenerV3Update(ctx context.Context, d *schema.ResourceData, meta
...
+	// Wait for LoadBalancer to become active again before continuing
+	err = waitForElbV3LoadBalancer(ctx, elbClient, loadBalancerID, "ACTIVE", nil, timeout)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
 func resourceListenerV3Delete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_listener.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go [hunk 21]
```diff
@@ -232,7 +232,7 @@ func resourceLoadBalancerV3Create(ctx context.Context, d *schema.ResourceData, m
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go [hunk 22]
```diff
@@ -352,21 +352,21 @@ func resourceLoadBalancerV3Read(_ context.Context, d *schema.ResourceData, meta
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go [hunk 23]
```diff
@@ -419,115 +419,15 @@ func resourceLoadBalancerV3Update(ctx context.Context, d *schema.ResourceData, m
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go [hunk 24]
```diff
@@ -556,24 +455,139 @@ func resourceLoadBalancerV3Update(ctx context.Context, d *schema.ResourceData, m
...
+	// Wait for LoadBalancer to become active before continuing
+	err = waitForElbV3LoadBalancer(ctx, elbClient, d.Id(), "ACTIVE", nil, timeout)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
 func resourceLoadBalancerV3Delete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_loadbalancer.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtank.go [hunk 25]
```diff
@@ -82,15 +82,15 @@ func resourceLogTankRead(_ context.Context, d *schema.ResourceData, meta interfa
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_logtank.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_logtank.go [hunk 26]
```diff
@@ -139,13 +139,13 @@ func resourceLogTankDelete(_ context.Context, d *schema.ResourceData, meta inter
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_logtank.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_member.go [hunk 27]
```diff
@@ -91,7 +91,7 @@ func resourceMemberV3Create(ctx context.Context, d *schema.ResourceData, meta in
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_member.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_member.go [hunk 28]
```diff
@@ -122,7 +122,7 @@ func resourceMemberV3Read(_ context.Context, d *schema.ResourceData, meta interf
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_member.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_member.go [hunk 29]
```diff
@@ -151,7 +151,7 @@ func resourceMemberV3Update(ctx context.Context, d *schema.ResourceData, meta in
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_member.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_monitor.go [hunk 30]
```diff
@@ -79,9 +79,9 @@ func ResourceMonitorV3() *schema.Resource {
 
 func resourceMonitorV3Create(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_monitor.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_monitor.go [hunk 31]
```diff
@@ -108,12 +108,12 @@ func resourceMonitorV3Create(ctx context.Context, d *schema.ResourceData, meta i
 
 func resourceMonitorV3Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_monitor.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_monitor.go [hunk 32]
```diff
@@ -147,7 +147,7 @@ func resourceMonitorV3Read(_ context.Context, d *schema.ResourceData, meta inter
 
 func resourceMonitorV3Update(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_monitor.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_monitor.go [hunk 33]
```diff
@@ -183,13 +183,13 @@ func resourceMonitorV3Update(ctx context.Context, d *schema.ResourceData, meta i
 
 func resourceMonitorV3Delete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_monitor.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_pool.go [hunk 34]
```diff
@@ -125,7 +125,7 @@ func resourcePoolV3Create(ctx context.Context, d *schema.ResourceData, meta inte
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_pool.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_pool.go [hunk 35]
```diff
@@ -171,7 +171,7 @@ func resourcePoolV3Read(_ context.Context, d *schema.ResourceData, meta interfac
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_pool.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_pool.go [hunk 36]
```diff
@@ -226,7 +226,7 @@ func resourcePoolV3Update(ctx context.Context, d *schema.ResourceData, meta inte
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_pool.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_pool.go [hunk 37]
```diff
@@ -268,7 +268,7 @@ func resourcePoolV3Delete(ctx context.Context, d *schema.ResourceData, meta inte
...
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_pool.go]: ditto

File: huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go [hunk 38]
```diff
@@ -256,14 +256,14 @@ func resourceSecurityPoliciesV3Update(ctx context.Context, d *schema.ResourceDat
 	config := meta.(*config.Config)
```
Comment (chengxiangdong) [inline huaweicloud/services/elb/resource_huaweicloud_elb_security_policy.go]: ditto

**Review by chengxiangdong — COMMENTED** [review]
It contains too many providers, please split it into multiple PRs by provider.

File: huaweicloud/services/elb/data_source_huaweicloud_elb_certificate.go (hunk 1 above)
Comment (houpeng80) [reply huaweicloud/services/elb/data_source_huaweicloud_elb_certificate.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go (hunk 3 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go (hunk 5 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_certificate.go]: done

File: huaweicloud/services/elb/resource_huaweicloud_elb_pool.go (hunk 34 above)
Comment (houpeng80) [reply huaweicloud/services/elb/resource_huaweicloud_elb_pool.go]: done

**Comment by chengxiangdong** [comment]
LGTM, 
UT results are also required, which may be performed a few days later.

**Pull request merged**